partner_multi_relation/models/res_partner_relation_all.py

```diff
@@ -0,0 +1,482 @@
+        string="Is reverse type?",
+        readonly=True,
+        help="Inverse relations are from right to left partner.",
+    )
+    type_selection_id = fields.Many2one(
+        comodel_name="res.partner.relation.type.selection",
+        string="Relation Type",
+        required=True,
+    )
+    active = fields.Boolean(
+        readonly=True,
+        help="Records with date_end in the past are inactive",
+    )
+    any_partner_id = fields.Many2many(
+        comodel_name="res.partner",
+        string="Partner",
+        compute=lambda self: self.update({"any_partner_id": None}),
+        search="_search_any_partner_id",
+    )
+
+    def register_specification(self, register, base_name, is_inverse, select_sql):
+        _last_key_offset = register["_lastkey"]
+        key_name = base_name + (is_inverse and "_inverse" or "")
+        assert key_name not in register
+        assert "%%(padding)s" in select_sql
+        assert "%(key_offset)s" in select_sql
+        assert "%(is_inverse)s" in select_sql
+        _last_key_offset += 1
+        register["_lastkey"] = _last_key_offset
+        register[key_name] = dict(
+            base_name=base_name,
+            is_inverse=is_inverse,
+            key_offset=_last_key_offset,
+            select_sql=select_sql
+            % {
+                "key_offset": _last_key_offset,
+                "is_inverse": is_inverse,
+                "extra_additional_columns": self._get_additional_relation_columns(),
+            },
+        )
+
+    def get_register(self):
+        register = collections.OrderedDict()
+        register["_lastkey"] = -1
+        self.register_specification(register, "relation", False, RELATIONS_SQL)
+        self.register_specification(register, "relation", True, RELATIONS_SQL_INVERSE)
+        return register
+
+    def get_select_specification(self, base_name, is_inverse):
+        register = self.get_register()
+        key_name = base_name + (is_inverse and "_inverse" or "")
+        return register[key_name]
+
+    def _get_statement(self):
+        """Allow other modules to add to statement."""
+        register = self.get_register()
+        union_select = " UNION ".join(
+            [register[key]["select_sql"] for key in register if key != "_lastkey"]
+        )
+        return """\
+        CREATE OR REPLACE VIEW %%(table)s AS
+             WITH base_selection AS (%(union_select)s)
+         SELECT
+             bas.*,
+             CASE
+                 WHEN NOT bas.is_inverse OR typ.is_symmetric
+                 THEN bas.type_id * 2
+                 ELSE (bas.type_id * 2) + 1
+             END as type_selection_id,
+             (bas.date_end IS NULL OR bas.date_end >= current_date) AS active
+             %%(additional_view_fields)s
+         FROM base_selection bas
+         JOIN res_partner_relation_type typ ON (bas.type_id = typ.id)
+         %%(additional_tables)s
+                """ % {"union_select": union_select}  # noqa: UP031
```

Comment:
noqa: UP031
I think you should replace 
`WITH base_selection AS (%(union_select)s)`
with
` WITH base_selection AS { union_select }`

Reply:
This didn't seem to work and I tried before to fix this properly. But no idea at the moment how to fix this.

Reply:
this should work:
```
return f"""\
        CREATE OR REPLACE VIEW %%(table)s AS
             WITH base_selection AS ({union_select})
         SELECT
             bas.*,
             CASE
                 WHEN NOT bas.is_inverse OR typ.is_symmetric
                 THEN bas.type_id * 2
                 ELSE (bas.type_id * 2) + 1
             END as type_selection_id,
             (bas.date_end IS NULL OR bas.date_end >= current_date) AS active
             %%(additional_view_fields)s
         FROM base_selection bas
         JOIN res_partner_relation_type typ ON (bas.type_id = typ.id)
         %%(additional_tables)s
                """
```

Reply:
I already tried that, but during compilation it gives me the following error:

```
psycopg2.errors.SyntaxError: syntax error at or near "%"
LINE 1:             CREATE OR REPLACE VIEW %(table)s AS
```

I tried many other things, but eventually gave up on the string formatting.

Reply:
Thanks for the help! Couldn't figure this one out myself :+1: